python/test/unit/language/test_conversions.py

```diff
@@ -253,17 +260,32 @@ def upcast_test(src_dtype, dst_dtype, exponent_bits, mantissa_bits, exponent_bia
     ('float8e4nv', 'float16'),
     ('float8e4nv', 'bfloat16'),
     ('float8e4nv', 'float32'),
+
+    ('float8e4b8', 'float32'),
+    ('float8e4b8', 'float16'),
+
+    ('float8e5b16', 'float32'),
+    ('float8e5b16', 'float16'),
 ])
 def test_typeconvert_upcast(src_dtype, dst_dtype):
 
-    if src_dtype == 'float8e4nv' and torch.cuda.get_device_capability(0) < (9, 0):
-        pytest.skip("float8e4nv upcast tests only supported on compute capability 9.0+")
+    if src_dtype == 'float8e4nv' and (torch.cuda.get_device_capability(0) < (9, 0)
+            or torch.version.hip is not None):
+        pytest.skip("float8e4nv upcast tests only supported on NVGPU with compute capability 9.0+")
+
+    if src_dtype in ['float8e4nv', 'float8e4b15'] and torch.version.hip is not None:
+        pytest.skip(f"{src_dtype} upcast tests not supported on ROCm")
+
+    if (src_dtype in ['float8e4b8', 'float8e5b16']) and (torch.version.hip is None or torch.cuda.get_device_capability(0) != (9, 4) ):
```

Comment:
why `torch.cuda.get_device_capability(0) != (9, 4)`? Should this be checking the hip target?

Reply:
On AMDGPU, `torch.cuda.get_device_capability(0)` will return part of the hip target. For examples, on the target "gfx942", it will return `(9, 4)`. I know this is a bit clumsy, as it will not be able to differentiate between some other targets like "gfx90a" and "gfx908" (both will return `(9, 0)`). There is some other way to get to know the target from torch, which from `torch.cuda.get_device_properties(0)`. For example, on MI210, it gives:
`_CudaDeviceProperties(name='AMD Instinct MI210', major=9, minor=0, total_memory=65520MB, multi_processor_count=104)`.

But I am a bit reserved on using the name string here to differentiate and would like to see if we can explore some better way to tell the hip target. For now, I am using `torch.cuda.get_device_capability(0)` for this PR as it would differentiate MI300 with other devices.

Reply:
I'm still seeing `torch.version.hip is None` being in the source code here?